app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -1,14 +1,14 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   include Devise::Controllers::Rememberable
 
-  # Called upon successful redirect from Twitter
-  def twitter
-    callback_for(:twitter)
-  end
+  skip_before_action :verify_authenticity_token, only: %i[apple] # rubocop:disable Rails/LexicallyScopedActionFilter
```

Comment:
This may be the main caveat of the PR. I don't think disabling this for Apple auth should be a problem but there may be a more elegant solution (I believe Apple sends in the data within a signed JWT). Anyways this is the reason behind having to open this

__Problem:__ Apple uses POST for the OAuth callback and because of this Rails "protects against forgery" from a different Origin and raises a CSRF exception. If you enable CORS debug you'll see this in the logs

```
HTTP Origin header (https://appleid.apple.com) didn't match request.base_url (https://rpi-forem.ngrok.io)
```

__Solution 1:__ Keep this `skip_before_action` for Apple's callback to avoid the CSRF exception

__Solution 2:__ Edit `config/initializers/cors.rb` in order to add an exception for this endpoint, only allowing `https://appleid.apple.com` instead of bypassing all checks. This is the equivalent to "_giving Apple the keys to enter through this very specific door_" vs "_leave the door open for anyone to come through_"

Reply:
Love the CORS solution, I'm still baffled as to their choice but everything about implementing this provider has been "slightly different" than the two previous ones